cwms-data-api/src/main/java/cwms/cda/data/dto/watersupply/WaterSupplyPump.java

```diff
@@ -68,28 +61,10 @@ public PumpType getPumpType() {
         return this.pumpType;
     }
 
-    public CwmsId getPumpId() {
-        return this.pumpId;
-    }
-
-    public void setPumpLocation(Location pumpLocation) {
-        this.pumpLocation = pumpLocation;
-    }
-
-    public void setPumpType(PumpType pumpType) {
-        this.pumpType = pumpType;
-    }
-
-    public void setPumpId(CwmsId pumpId) {
-        this.pumpId = pumpId;
-    }
-
     @Override
-    public void validate() {
-        this.pumpLocation.validate();
-        if (this.pumpType == null) {
-            throw new FieldException("Pump Type cannot be null");
-        }
-        this.pumpId.validate();
+    protected void validateInternal(CwmsDTOValidator validator) {
+        super.validateInternal(validator);
+        validator.required(getPumpLocation(), "pump-location");
```

Comment:
use @JsonProperty(required=true) on the class variable instead of implementing validateInternal